### PR TITLE
Refactor Error Handling in Exposure Submission

### PR DESCRIPTION
**Pull request by melloskitten**
### EDIT:

As decided with the team, this PR will **not** change any error messages or error texts.  Instead, it is introducing a refactoring for the error parsing in the Exposure Submission flow. Thanks for the understanding.
___

~~This PR is part of the [improved error handling BLI (#1552)](https://jira.itc.sap.com/browse/EXPOSUREAPP-1552), namely solving this issue [here (#1628)](https://jira.itc.sap.com/browse/EXPOSUREAPP-1628).~~

~~I have added the missing fields `.internal, .unsupported and .rateLimited`, as mentioned in ErrorMessageCollection.xlsx (If you need the link to the excel sheet, drop me a message). Please note that these only included the **german localization** of the error messages.~~

~~@steins-code, could you perhaps check whether there are corresponding keys for the new entries on the other Localization files, please? Thank you.~~

### CODE REVIEW

File: src/xcode/ENA/ENA/Resources/Localization/de.lproj/Localizable.strings [hunk 1]
```diff
@@ -477,6 +477,12 @@
 
 "ExposureSubmissionError_defaultError" = "Allgemeiner Exposure Submission Fehler";
 
+"ExposureSubmissionError_internal" = "Fehler 11. Es ist ein interner Fehler passiert. Bitte starten Sie das Telefon neu und probieren Sie es morgen erneut.";
```
Comment (steins-code) [inline src/xcode/ENA/ENA/Resources/Localization/de.lproj/Localizable.strings:480]: Can you add an intermediate english translation for those and put these also into turkish language file?;)

**Comment by melloskitten** [comment]
As decided by the team, we will NOT add refined error messages to this PR.
However, as I refactored quite a bit of code, I will update this PR to only contain refactorings, and remove anything related to the additional error messages.

**Review by haosap — APPROVED** [review]

File: src/xcode/ENA/ENA/Source/Extensions/ExposureSubmissionParsable.swift [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+import ExposureNotification
+
+// MARK: - ExposureSubmissionErrorTransformable protocol.
+
+/// This protocol ensures that a given ErrorType can be transformed into an
+/// `ExposureSubmissionError`.
+/// For the future, if other transformations are needed, it is advised to create
+/// a corrseponding protocol specific to the destination error type.
+protocol ExposureSubmissionErrorTransformable {
+	func transformToExposureSubmissionError() -> ExposureSubmissionError
```
Comment (johannesrohwer) [inline src/xcode/ENA/ENA/Source/Extensions/ExposureSubmissionParsable.swift:30]: maybe drop the "transform" prefix and just go with `toExposureSubmissionError`? Makes the API a bit shorter without losing any information

Comment (melloskitten) [reply src/xcode/ENA/ENA/Source/Extensions/ExposureSubmissionParsable.swift:30]: will do.

**Review by johannesrohwer — APPROVED** [review]

**Pull request merged**